eox_core/api/v1/views.py

```diff
@@ -695,20 +695,74 @@ class EdxappGrade(UserQueryMixin, APIView):
     permission_classes = (EoxCoreAPIPermission,)
     renderer_classes = (JSONRenderer, BrowsableAPIRenderer)
 
+    @apidocs.schema(
+        parameters=[
+            apidocs.query_parameter(
+                name="username",
+                param_type=str,
+                description="**required**, The username used to identify a user enrolled on the course. Use either username or email.",
+            ),
+            apidocs.query_parameter(
+                name="email",
+                param_type=str,
+                description="**required**, The email used to identify a user enrolled on the course. Use either username or email.",
+            ),
+            apidocs.query_parameter(
+                name="course_id",
+                param_type=str,
+                description="**required**, The course id for the enrollment you want to check.",
+            ),
+            apidocs.query_parameter(
+                name="detailed",
+                param_type=bool,
+                description="**optional**, If true include detailed data for each graded subsection",
+            ),
+            apidocs.query_parameter(
+                name="grading_policy",
+                param_type=bool,
+                description="**optional**, If true include course grading policy.",
+            ),
+        ],
+        responses={
+            200: EdxappGradeSerializer,
+            400: "Bad request, missing course_id or either email or username",
+            404: "User, course or enrollment not found",
+        },
+    )
     def get(self, request):
         """
         Retrieves Grades information for given a user and course_id
 
         **Example Requests**
 
-            GET /eox-core/api/v1/grade/?username=johndoe&
-            course_id=course-v1:edX+DemoX+Demo_Course
+            GET /eox-core/api/v1/grade/?username=johndoe&course_id=course-v1:edX+DemoX+Demo_Course
 
             Request data: {
               "username": "johndoe",
               "course_id": "course-v1:edX+DemoX+Demo_Course",
             }
 
+        **Response details**
+
+        - `earned_grade`: Final user score for the course.
+        - `section_breakdown`: Details of each grade subsection.
```

Comment:
Can we put in this place that this is optional?  would that break the format expected by drf-yasg?

Reply:
This is part is just markdown, so adding a little text saying is optional would be no problem.